### DISCUSSION TITLE
Run xUnit using GitHub Action in a .NetFramework project.

### DISCUSSION THREAD

**Discussion by sonomirco** [community/community#45116] [Actions] [answered]
### Select Topic Area

Question

### Body

Hi there, 
Could you please help me with this task?
I tried setting a target, as explained here
https://xunit.net/docs/getting-started/netfx/cmdline

But I need to add it to all the other projects I have in the solution, and after I tried, this is what I got.
![image](https://user-images.githubusercontent.com/57708659/213977846-324fcf71-1483-49bd-b253-9435a9250ac6.png)

Because I am using.
![image](https://user-images.githubusercontent.com/57708659/213977957-de0e1b0d-ad04-453a-98b5-e2bc073b4d75.png)

Could it be a solution using this action?
https://github.com/microsoft/vstest-action

The process will be as follows:
1. restore the package
2. build the solution
3. use `vstest-action` with the specific test assembly.

Any other suggestions are welcome.
Thanks

**Comment by jge162** [comment]
@sonomirco try to use vstest.console.exe command line tool to run the test by specifying the assembly path 

vstest.console.exe <path-to-test-assembly>

**Reply by jge162** [reply]
Code blocks not working. See screen grab below.

**Reply by sonomirco** [reply]
@jge162 I don't see the screenshot but if I can run the command without the need a specific action to install vstest even better.
How could I capture the results running vstest?

**Reply by jge162** [reply]
This will generate a trx file in the same directory as the test assembly, which you can then use to view the test results.

```CLI
vstest.console.exe MyTestAssembly.dll /logger:trx
```

you can also use the

```CLI
/Logger:console;verbosity=normal 
```
switch, which will print the results to the console in real-time.

**Reply by sonomirco** [reply]
I'll try as soon as I can.

**Reply by jge162** [reply]
Np. Should work. :)

**Reply by sonomirco** [reply]
Ok, I was able to test and it doesn't work just passing the command.
    - name: 🧪 Test
      run: vstest.console.exe ${{ env.TEST_BUILD }} /Logger:trx
      
This is the error.
![image](https://user-images.githubusercontent.com/57708659/214026451-11fa5144-0c42-4cfe-896c-9541c1460fb7.png)

**Reply by jge162** [reply]
Ensure that the environment variable TEST_BUILD is set correctly and that it contains the path to the test build you want to run.

Verify that the necessary dependencies, such as the .NET Framework, are installed on the machine where the command is being executed.

Try running the command with the /diag option to generate more detailed diagnostic output that may give more information about what's causing the error.

**Reply by jge162** [reply]
Make sure that the vstest.console.exe is installed in your machine, you can download it in the official Microsoft website.

**Reply by jge162** [reply]
The "squiggly lines" or "squiggly red lines" you are seeing under vstest.console.exe could indicate that the file or path cannot be found, or that there is an issue with your environment variables.

**Reply by sonomirco** [reply]
The "squiggly red lines" tells me vstest.console.exe is not installed.
I am running this on a GitHub action, so I assume vstest.console.exe needs to be installed as is explained here.
https://dev.to/felipetofoli/github-actions-for-net-full-framework-build-and-test-299h

**Reply by jge162** [reply]
One way to install vstest.console.exe on a GitHub action is by adding steps to your GitHub action workflow that install the necessary dependencies. for example:


- name: Install vstest.console.exe
   run: |
    choco install
    vstest.console.exe

There is a dash (-) in front of - name install. Can’t do code block. 

You can use similar steps to install other dependencies, such as the .NET Framework, that are required to run your tests.

but refer to the link you sent above.

**Reply by sonomirco** [reply]
I'll try to install vstest.console.exe using choco will avoid to use a specific GitHub action as shown in the link.

**Reply by jge162** [reply]
Ok should be good. Because Using Chocolatey to install vstest.console.exe is a good option as it allows for easy and automated installation of the package. However, you will need to make sure that Chocolatey is installed on your system first before you can use it to install vstest.console.exe.

**Reply by sonomirco** [reply]
Yep, it should be good.

Chocolatey is the package manager of choice for Windows, and it's installed and ready to use in the GitHub Actions virtual environments.

**Reply by jge162** [reply]
Sweet. Ok gn.

**Reply by sonomirco** [reply]
I’ll let you know.
I can’t find any about installing vstest via choco, but let’s see!

**Reply by jge162** [reply]
You can also use a GitHub action from the marketplace that already provides vstest.console.exe, like "Azure DevOps Test"

**Reply by jge162** [reply]
You can download the vstest.console.exe binary and include it in your repository. Then you can reference it in your GitHub Action workflow and run the tests using vstest.console.exe.

**Comment by sonomirco** [comment]
`
name: Install vstest.console.exe
run: choco install vstest.console.exe
`
As I thought this didn't work because it doesn't exist.
![image](https://user-images.githubusercontent.com/57708659/214157405-f5122ef2-ee47-4f73-93d1-4fd901ec3877.png)

I will use the GitHub actions used in this post.
https://dev.to/felipetofoli/github-actions-for-net-full-framework-build-and-test-299h

**Reply by jge162** [reply]
Sorry, I tried mention in my last post you had to download it. I realized after Choco did not have it...

**Reply by sonomirco** [reply]
I used the process shown in the post, adding the creation report at the end.

**Reply by jge162** [reply]
sweet, nice to have meet you.

**Comment by queenofcorgis** [comment, marked as answer]
Thanks for posting in the GitHub Community, @sonomirco ! 

We're happy you're here. You are more likely to get a useful response if you are posting your question in the applicable category, the Issues category is solely related to conversations around the GitHub products Issues and Projects. I've gone ahead and moved it for you. Good luck!
